torch/csrc/utils/tensor_new.cpp

```diff
@@ -463,13 +465,16 @@ Tensor tensor_ctor(const Type& type, PyObject* args, PyObject* kwargs) {
   auto r = parser.parse(args, kwargs, parsed_args);
   if (r.idx == 0) {
     bool type_inference = r.isNone(1);
+    // args_requires_grad=True only if requires_grad is set and is set to True (requires_grad=True)
+    bool args_requires_grad = !r.isNone(3) && r.toBool(3);
```

Comment:
I'm surprised that with the default specification `bool requires_grad=False` you need to test the `isNone` case...